homeassistant/components/oasa_telematics/sensor.py

```diff
@@ -0,0 +1,201 @@
+    stop_id = config[CONF_STOP_ID]
+    route_id = config.get(CONF_ROUTE_ID)
+
+    data = OASATelematicsData(stop_id, route_id)
+
+    add_devices([OASATelematicsSensor(
+        data, stop_id, route_id, name)], True)
+
+
+class OASATelematicsSensor(Entity):
+    """Implementation of the OASA Telematics sensor."""
+
+    def __init__(self, data, stop_id, route_id, name):
+        """Initialize the sensor."""
+        self.data = data
+        self._name = name
+        self._stop_id = stop_id
+        self._route_id = route_id
+        self._name_data = self._times = self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        params = {}
+        if self._times is not None:
+            next_arrival = None
+            if len(self._times) > 1:
+                next_arrival = ('{} min'.format(
+                    str(self._times[1][ATTR_DUE_IN])))
+            params.update({
+                ATTR_DUE_IN: str(self._times[0][ATTR_DUE_IN]),
+                ATTR_ROUTE_ID: self._times[0][ATTR_ROUTE_ID],
+                ATTR_STOP_ID: self._stop_id,
+                ATTR_ATTRIBUTION: ATTRIBUTION,
+                ATTR_NEXT_ARRIVAL: next_arrival
+            })
+        params.update({
+            ATTR_ROUTE_NAME: self._name_data[ATTR_ROUTE_NAME],
+            ATTR_STOP_NAME: self._name_data[ATTR_STOP_NAME]
+        })
+        return {k: v for k, v in params.items() if v}
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return 'min'
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON
+
+    def update(self):
+        """Get the latest data from OASA API and update the states."""
+        self.data.update()
+        self._times = self.data.info
+        self._name_data = self.data.name_data
+        try:
+            self._state = self._times[0][ATTR_DUE_IN]
+        except TypeError:
+            pass
+
+
+class OASATelematicsData():
+    """The class for handling data retrieval."""
+
+    def __init__(self, stop_id, route_id):
+        """Initialize the data object."""
+        import oasatelematics
+        self.stop_id = stop_id
+        self.route_id = route_id
+        self.info = self.empty_result()
+        self.name_data = self.empty_name_data()
+        self.oasa_api = oasatelematics
+
+    def empty_result(self):
+        """Object returned when no arrivals are found."""
+        return [{ATTR_DUE_IN: 'n/a',
+                 ATTR_ROUTE_ID: self.route_id}]
+
+    @staticmethod
+    def empty_name_data():
+        """Object returned when no stop/route name data are found."""
+        return [{ATTR_STOP_NAME: 'n/a',
```

Comment:
Instead of setting `'n/a'`, don't include these attributes in the state attributes at all if not known. If state isn't known we should set state to `None`.